client/packages/system/src/Item/api/operations.graphql

```diff
@@ -35,8 +35,12 @@ fragment ItemRow on ItemNode {
   unitName
 }
 
-fragment ItemRowWithStats on ItemNode {
+fragment ItemWithPackSize on ItemNode {
   ...ItemRow
+  defaultPackSize
```

Comment:
I am a little bit confused here, maybe defaultPackSize should be on ItemRow ?

Reply:
I think it would reduce the diff quite a bit ?

Reply:
if I put it on `ItemRow` then I have to update a ton more files - lots of things use the ItemRow fragment. and then I have to update all the queries for parent objects too, so that I can pass the correct shape down when selecting items.

I'm also concerned at the approach of just adding more stuff on to item, and making that object huge when some properties are only used in specific places. Want to keep the set of properties requested as small as possible.

Reply:
> and then I have to update all the queries for parent objects too

I would have thought we use fragments within the query to avoid changing queries when fragment changes (I thought that's what the intended use case for fragments).

Anyways I take your point about queries being specific, in that case can we use query result types vs creating fragments (just something for the future): https://github.com/openmsupply/open-msupply/compare/354-inbound-changes...354-example

Reply:
the use of result types is nice! thanks - I've updated the inbound uses and I think it's clearer.

it's not updating the queries that's required - sorry, that was an unclear comment. typescript raises issues if the shape of `ItemRow` changes, as we have a lot of methods that are using shortcuts, like spreading an object, that can't be satisified if the destination object now has more properties on it. Some of that requires fetching more props from the original query in order to pass them around.. and things generally got out of hand.